lib/kubeclient/common.rb

```diff
@@ -68,6 +68,7 @@ def handle_exception
         {}
       end
       err_message = json_error_msg['message'] || e.message
+      raise NameError.new if json_error_msg && json_error_msg['reason'] == 'NotFound'
```

Comment:
@moolitayer isn't "NotFound" used also when you look for a specific entity and it's not found? (e.g. `get_pod("my_pod")`).